Source/astcenc_compress_symbolic.cpp

```diff
@@ -306,9 +309,10 @@ static void compress_symbolic_block_fixed_partition_1_plane(
 	int qwt_bitcounts[MAX_WEIGHT_MODES];
 	float qwt_errors[MAX_WEIGHT_MODES];
 
-	for (int i = 0; i < MAX_WEIGHT_MODES; i++)
+	for (int i = 0, ni = bsd->block_mode_packed_count; i < ni; ++i)
```

Comment:
I kinda got used to this pattern to "make sure" that the compiler does not decide to do a load of `bsd->block_mode_packed_count` on each iteration of the loop (which it may or might not realize is possible)